pkg/prom/prom.go

```diff
@@ -374,6 +378,21 @@ type PrometheusClientConfig struct {
 
 // NewPrometheusClient creates a new rate limited client which limits by outbound concurrent requests.
 func NewPrometheusClient(address string, config *PrometheusClientConfig) (prometheus.Client, error) {
+
+	var tlsCaCert *x509.CertPool
+	// get the bearer token and ca cert from serviceaccount if we have kube rbac proxy enabled
+	if env.IsKubeRbacProxyEnabled() {
+		restConfig, err := restclient.InClusterConfig()
+		if err != nil {
+			log.Fatalf("Failed to get in-cluster config: %s", err)
```

Comment:
Probably inappropriate to perform a `log.Fatalf` here? This would terminate the whole application. We should probably just return the error? The error message should be descriptive stating that `ENABLE_KUBE_RBAC_PROXY` was set to true but we couldn't retrieve the Bearer Token or Cert?

Reply:
my bad here, thanks for pointing this out. updating it